### PR TITLE
MdeModulePkg/DxeCapsuleLibFmp: Fix crash if no ESRT is found

### PR DESCRIPTION
**Pull request by nhivp**
# Description

The ESRT table is not required in UEFI firmware. In such cases, the table may not be present in the UEFI Configuration Table. The mEsrtTable is to check if the IsNestedFmpCapsule() function is invoked at runtime to determine whether to use gEsrtManagementProtocolGuid or the ESRT table from the Configuration Table. Unfortunately, the check does not cover situations where the ESRT is not present, potentially resulting in a kernel crash. This patch is intended to fix this issue.

- [ ] Breaking change?
  - **Breaking change** - Will this cause a break in build or boot behavior?
  - Examples: Add a new library class or move a module to a different repo.
- [ ] Impacts security?
  - **Security** - Does the change have a direct security impact?
  - Examples: Crypto algorithm change or buffer overflow fix.
- [ ] Includes tests?
  - **Tests** - Does the change include any explicit test code?
  - Examples: Unit tests or integration tests.

## How This Was Tested

<_Describe the test(s) that were run to verify the changes._>

## Integration Instructions

N/A

### CODE REVIEW

File: MdeModulePkg/Library/DxeCapsuleLibFmp/DxeCapsuleLib.c [hunk 1]
```diff
@@ -1402,6 +1405,16 @@ IsNestedFmpCapsule (
       }
     }
   } else {
+    if (mDxeCapsuleLibReadyToBootEvent) {
```
Comment (mdkinney) [inline MdeModulePkg/Library/DxeCapsuleLibFmp/DxeCapsuleLib.c:1408]: The last time gBS->LocateProtocol() can be called is ExitBootServices().  Are there cases where this API is used between ReadyToBoot and ExitBootServices?

**Comment by nhivp** [comment]
@lgao4 Could you review, please?

File: MdeModulePkg/Library/DxeCapsuleLibFmp/DxeCapsuleLib.c (hunk 1 above)
Comment (lgao4) [reply MdeModulePkg/Library/DxeCapsuleLibFmp/DxeCapsuleLib.c:1408]: I don't see the case that gBS->LocateProtocol () is called at runtime. Could you introduce it?

Comment (nhivp) [reply MdeModulePkg/Library/DxeCapsuleLibFmp/DxeCapsuleLib.c:1408]: The `mEsrtTable` is NULL if the ESRT is not located at ReadyToBoot. During runtime, when `IsNestedFmpCapsule()` is called, it defaults to the 'else' case, resulting in a crash.

Comment (lgao4) [reply MdeModulePkg/Library/DxeCapsuleLibFmp/DxeCapsuleLib.c:1408]: I agree this problem. This fix is good.

**Review by lgao4 — APPROVED** [review]

**Pull request merged**